src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -226,7 +226,8 @@ private void DetermineSerializationCapabilities()
                         Type elementType = JsonClassInfo.GetElementType(RuntimePropertyType, ParentClassType, PropertyInfo);
 
                         // If the property type only has interface(s) exposed by JsonEnumerableT<T> then use JsonEnumerableT as the converter.
```

Comment:
For performance, these interface-only converters we should be able to just use `List<T>` instead of wrapping with `JsonEnumerableT<T>`. I did that for `[IDictionary<>]`(https://github.com/dotnet/corefx/blob/94c5f1384c9b6477d164050de964cf6dcf99b2a9/src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs#L91)

p.s. I know at some point in the past we were concerned about "leaking" the implementation, but I don't think that is a concern any longer.